tools/Dashboard/DevHome.Dashboard/Helpers/WidgetHelpers.cs

```diff
@@ -67,10 +70,18 @@ public static double GetPixelHeightFromWidgetSize(WidgetSize size)
         };
     }
 
-    public static bool IsIncludedWidgetProvider(WidgetProviderDefinition provider)
+    public static async Task<bool> IsIncludedWidgetProviderAsync(WidgetProviderDefinition provider)
     {
-        var include = provider.Id.StartsWith("Microsoft.Windows.DevHome", StringComparison.CurrentCulture);
-        Log.Logger()?.ReportInfo("WidgetHelpers", $"Found provider Id = {provider.Id}, include = {include}");
+        var providerId = provider.Id;
+
+        var extensionService = Application.Current.GetService<IExtensionService>();
+        var enabledWidgetProviderIds = await extensionService.GetInstalledDevHomeWidgetPackageFamilyNamesAsync(true);
+
+        var endOfPfnIndex = providerId.IndexOf('!', StringComparison.Ordinal);

```

Comment:
Is the `GetInstalledDevHomeWidgetPackageFamilyNamesAsync()` method the appropriate place for this task? As far as I know, package family names do not include `!`.

Reference: https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/package-identity-overview#package-full-name

Reply:
Maybe the order this code is in is confusing... we are returning PackageFamilynames, but the ProviderDefinition's Id is PackageFamilyName!AppInfo.Id!!AppExtension.Id (the same as our ExtensionUniqueIds). That we have to trim down to compare to DevHome extensions' PFNs.

Reply:
Reordered and added comments.